### PR TITLE
BUG: translate losing object dtype with new string dtype

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/strings/accessor.py [hunk 1]
```diff
@@ -2317,7 +2318,8 @@ def translate(self, table):
         dtype: object
         """
         result = self._data.array._str_translate(table)
-        return self._wrap_result(result)
+        dtype = object if self._data.dtype == "object" else None
+        return self._wrap_result(result, dtype=dtype)
```
Comment (mroeschke) [inline pandas/core/strings/accessor.py:2322]: Maybe dumb question, but we can't just pass `dtype=self._data.dtype` and have the result dtype be correct in `_wrap_result`?

Comment (phofl) [reply pandas/core/strings/accessor.py:2322]: Categoricals act out then, that's a bit confusing to me as well, but I didn't want to focus on this now.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Comment by phofl** [comment]
Sorry this was meant for 2.1.4, should have added the milestone

**Comment by phofl** [comment]
@meeseeksdev backport 2.1.x

**Pull request merged**